### PR TITLE
malus: add `finality_delay` cli flag

### PR DESCRIPTION
**Pull request by ordian**
This allows to pass a `--finality-delay X` cli param to malus to configure the finality delay.

~~The ugly part is that it's feature gated with `#[cfg(feature = "malus")]` in `new_full` function. However, due to how cargo resolves features, in some packages, `malus` is enabled when launching from workspace root, but not if individually (e.g. running only adder-collator tests). I'm open to ideas to how avoid that w/o exposing this flag to validators.~~ This param propagates to cumulus unfortunately, not sure if there's a sane way to hide it.

cumulus companion: https://github.com/paritytech/cumulus/pull/1442

### CODE REVIEW

**Review by sandreim — COMMENTED** [review]
I don't think it is a good idea to add any malus stuff in the real node. We should add a mechanism that allows us to feed any override into the node from the malus codebase. 

There is another CLI flag that might help with your testing - https://github.com/paritytech/polkadot/blob/decdbedea6d49b0604fee3241970a7ff383fe1db/cli/src/cli.rs#L112 Does this work ?

**Comment by ordian** [comment]
> I don't think it is a good idea to add any malus stuff in the real node.

The cli flag is now only available for malus binaries. As an extra protection, it's only activated if `malus` feature is enabled.

>  We should add a mechanism that allows us to feed any override into the node from the malus crate.

`new_full` is already a way to pass this override. Whether we pass a bunch of params or a closure that will take another bunch or params, it doesn't change that. We could create a wrapper for `build_full` for cumulus and non-malus node that doesn't expose malus-specific flags if that's helpful.

> There is another CLI flag that might help with your testing 

I'm not sure yet if `grandpa_pause` will be enough for my use case, but it's less flexible in that only delay finality once (which might make some of the tests more flaky).

File: cli/src/command.rs [hunk 1]
```diff
@@ -257,13 +257,18 @@ fn host_perf_check() -> Result<()> {
 /// accepts an alternative overseer generator, to adjust behavior
 /// for integration tests as needed.
 #[cfg(feature = "malus")]
-pub fn run_node(run: Cli, overseer_gen: impl service::OverseerGen) -> Result<()> {
-	run_node_inner(run, overseer_gen, |_logger_builder, _config| {})
+pub fn run_node(
```
Comment (sandreim) [inline cli/src/command.rs:260]: Nit: we could briefly document the new parameter and how the delay is applied.

**Review by sandreim — APPROVED** [review]

**Review by bkchr — APPROVED** [review]

**Comment by ordian** [comment]
bot merge

**Pull request merged**